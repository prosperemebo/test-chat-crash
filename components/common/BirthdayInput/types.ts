export interface BirthdayInputProps {
  error?: string
  value?: string | Date
  onChange: (value: string) => void
}