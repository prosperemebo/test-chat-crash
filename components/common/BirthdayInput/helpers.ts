export const formatBirthday = (date: string) => {
  const birthdayDate = new Date(date)
  return birthdayDate.toLocaleDateString('en-US', {
    year: 'numeric',
    month: 'short',
    day: '2-digit',
  })
}

export const formatDate = (date: Date): string => {
  return date.toLocaleDateString('en-US', {
    month: '2-digit',
    day: '2-digit',
    year: 'numeric',
  })
}

export const convertToDate = (dateStr: string) => {
  const [month, day, year] = dateStr.split('/')
  return new Date(`${year}-${month}-${day}T00:00:00`)
}