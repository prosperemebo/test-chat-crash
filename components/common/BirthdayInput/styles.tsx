import styled from 'styled-components/native'
import { TextInputMask } from 'react-native-masked-text'
import { StyleSheet } from 'react-native'

export const BirthdayTextInput = styled(TextInputMask)`
    font-size: 15px;
    font-family: 'Poppins-Regular';
    color: ${props => props.theme.common.inputTextColor};
    width: 100%;
`

export const ItemLabel = styled.Text`
  flex: 8;
  font-size: 15px;
  font-family: 'Poppins-Regular';
  margin-left: 0;
  padding-left: 0;
  color: ${props => props.theme.common.serviceTextColor};
`

export const BirthdayButton = styled.TouchableOpacity`
  flex-direction: row;
  align-items: center;
  border-radius: 8px;
  border-width: ${StyleSheet.hairlineWidth};
  border-color: ${props => props.theme.common.inputBorderColor};
  background-color: ${props => props.theme.common.inputBackgroundColor};
  width: 100%;
  padding: 15px;
`

export const DateInputContainer = styled.View`
  flex-direction: row;
  align-items: center;
  border-radius: 8px;
  border-width: ${StyleSheet.hairlineWidth}px;
  border-color: ${props => props.theme.common.inputBorderColor};
  background-color: ${props => props.theme.common.inputBackgroundColor};
  width: 100%;
  padding: 15px;
  margin-bottom: 5px;
`