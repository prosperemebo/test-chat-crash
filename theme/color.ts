export const WHITE = '#FFFFFF'
export const t10WHITE = 'rgba(255,255,255,0.1)'
export const t76WHITE = 'rgba(255,255,255,0.76)'

export const RED = '#f60004'
export const RED1 = '#D33E41'


export const BLACK = '#000000'
export const BLACK2 = '#050608'
export const nBLACK = '#15161A'
export const nBLACK10 = '#1F2025'
export const nBLACK30 = '#28292E'
export const nBLACK40 = '#050608'

export const nGRAY10 = '#6B6E79'
export const nGRAY20 = '#858995'
export const nGRAY25 = '#54575e'
export const nGRAY30 = '#2f3137'
export const nGRAY35 = '#52555D'
export const nGRAY40 = '#35363B'


export const tGRAY20 = '#85899599'
export const tGRAY30 = '#545458A6'
export const tGRAY40 = '#45464C'
export const GRAY40 = '#cfcfd0'
export const GRAY45 = '#c9c9cb'
export const GRAY50 = '#aaaaaa'
export const GRAY60 = '#52545D'
export const t40GRAY60 = '#52545D66'
export const GRAY70 = '#383B40'
export const GRAY90 = '#27292d'
export const GRAY100 = '#1f2023'
export const GRAY110 = '#151619'

export const GRAY_ONBOARD = '#8791a5'

export const BLUE = '#50b5e8'
export const BLUE2 = '#0A84FF'

export const PURPLE = '#9A1F99'

export const nBLUE = '#8D88A8'
export const nPURPLE = '#C47AF5'
export const nPURPLE10 = '#873FBB'
export const nPURPLE20 = '#873FBB'
export const nPURPLE40 = '#C882F8'

export const nORANGE10 = '#FF6705'
export const nORANGE20 = '#FF7905'
export const nORANGE40 = '#FFB700'

export const YELLOW2 = '#EBC789'
export const nYELLOW = '#F5D193'

export const nGRAYBACKGROUND2 = '#292A30'
export const nGRAYBACKGROUND3 = '#383940'
export const nGRAYBACKGROUND4 = '#989898'

export const gradient = {
  COLORS: [BLUE, PURPLE],
  START: { x: 0, y: 1 },
  END: { x: 1, y: 0 },
  LABEL_COLOR: WHITE,
}

export const orangeGradient = {
  COLORS: [nORANGE10, nORANGE40],
  START: { x: 0, y: 1 },
  END: { x: 1, y: 0 },
  LABEL_COLOR: WHITE,
}