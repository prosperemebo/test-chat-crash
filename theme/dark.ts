import {
  WHITE,
  BLACK,
  nGRAY10,
  GRAY_ONBOARD,
  nPURPLE,
  nBLUE,
  GRAY40,
  nGRAY20,
  RED1,
  tGRAY20,
  RED,
  nBLACK,
  nORANGE10,
  nORANGE40,
  nPURPLE10,
  GRAY70,
  tGRAY30,
  BLUE2,
  BLACK2,
  nGRAY25,
  nBLACK30,
  GRAY50,
  nBLACK10,
  t10WHITE,
  nGRAYBACKGROUND2,
  GRAY100,
  GRAY60,
  nGRAY40,
  nGRAY30,
  YELLOW2,
  nYELLOW,
  nGRAYBACKGROUND3,
  tGRAY40,
  nGRAY35,
  nBLACK40,
  t40GRAY60,
  nPURPLE20,
  nPURPLE40,
} from './color'

export const dark = {
  tint: nPURPLE,
  primaryTextColor: WHITE,
  primaryBg: nBLACK10,
  headerBg: GRAY100,
  headerBorderColor: GRAY60,
  secondaryTextColor: GRAY60,
  secondaryBgLight: nBLACK30,
  secondaryBg: nBLACK,
  serviceBg: GRAY100,
  inActiveBorderColor2: t10WHITE,
  backgroundColor2: nGRAYBACKGROUND2,
  backgroundColor3: nGRAYBACKGROUND3,
  backgroundColor4: nBLACK10,

  auth: {
    socialIconColor: WHITE,
    socialLoginButton: BLACK,
    dividerColor: nGRAY10,
    noticeText: WHITE,
    forgotPasswordLabel: WHITE,
  },

  onboarding: {
    headerTextColor: WHITE,
    headerStepsTextColor: GRAY_ONBOARD,
    progressBarColor: nPURPLE,
    progressBarBackgroundColor: nBLUE,
    selectedItemTextColor: WHITE,
    disabledItemTextColor: GRAY40,
    gameSectionOverlayColor: nPURPLE,
    gameEmptySectionColor: BLACK,
    gameEmptyIconColor: BLACK2,
  },

  common: {
    labelColor: WHITE,
    infoTextColor: nGRAY20,
    inputIconColor: nGRAY20,
    inputPlaceholderColor: nGRAY20,
    serviceTextColor: WHITE,
    inputTextColor: WHITE,
    inputTextError: RED1,
    inputBorderColor: tGRAY20,
    inputErrorBorderColor: RED,
    inputBackgroundColor: nBLACK,
    toastBackground: nBLACK,
    toastBorderColor: nPURPLE,
    toastTextColor: WHITE,
    buttonGradientOrangeStart: nORANGE10,
    buttonGradientOrangeEnd: nORANGE40,
    buttonGradientPurpleStart: nPURPLE40,
    buttonGradientPurpleEnd: nPURPLE20,
    buttonGradientGrayStart: nGRAY10,
    buttonGradientGrayEnd: nGRAY20,
    buttonDartBackgroundColor: BLACK,
    backButtonColor: nGRAY20,
    switchTextColor: WHITE,
    checkboxIconColor: nPURPLE,
    gradientButtonColorStart: nPURPLE10,
    gradientButtonColorEnd: nPURPLE,
    borderColor: GRAY70,
    genderPickerBorderColor: nGRAY25,
    genderPickerActiveTextColor: WHITE,
    genderPickerTextColor: GRAY50,
    checkboxTextColor: WHITE,
    iconButtonColor: WHITE,
    backgroundColor: nBLACK10,
  },

  settings: {
    socialIconBackgroundColor: nGRAY40,
    premiumFeatureTextColor: nPURPLE,
    buttonIconColor: nGRAY20,
    serviceItemBgColor: nGRAY30,
  },

  postOnboarding: {
    backgroundColor: nBLACK10,
    inputIconColor: nGRAY20,
    inputIconErrorColor: RED1,
    disclosureTextColor: nGRAY20,
    loadingIndicatorColor: BLACK,
  },

  profile: {
    backgroundColor: nBLACK10,
    improveBackgroundColor: YELLOW2,
    improveBannerBackgroundColor: nYELLOW,
    improveBannerTitleColor: nBLACK30,
    coverPictureCardRemoveButtonBg: tGRAY30,
    profilePicturesCountColor: tGRAY40,
  },

  chats: {
    iconColor: nGRAY20,
    inputColor: nBLACK40,
    placeholderColor: nGRAY35,
    appbarBorderColor: t40GRAY60,
    backgroundColor: nBLACK10,
  },

  nav: {
    bottomTabLabelBackgroundColor: nBLACK,
    bottomTabBackgroundColor: nBLACK,
  },

  other: {
    updateApp: {
      textBorderColor: tGRAY30,
      buttonTextColor: BLUE2,
      textColor: WHITE,
    },
  },
}
