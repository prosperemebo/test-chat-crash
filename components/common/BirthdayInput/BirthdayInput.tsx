import { useState } from 'react'
import { useTheme } from 'styled-components/native'
import i18n from '@/localization/i18n'
import { BirthdayTextInput, ItemLabel, BirthdayButton, DateInputContainer } from './styles'
import { InfoText } from '@/components/common'
import { BIRTHDAY_FORMAT } from '@/constants/general'
import { BirthdayInputProps } from './types'
import { convertToDate, formatBirthday, formatDate } from './helpers'

const BirthdayInput = (props: BirthdayInputProps) => {
  const { value, onChange, error } = props
  const theme = useTheme()

  const [isEditing, setIsEditing] = useState(false)
  const [inputValue, setInputValue] = useState(value ? formatDate(value as Date) : '')

  const handleTextChange = (text: string) => {
    setInputValue(text)
  }

  const handleEndEditing = () => {
    const formattedDate = convertToDate(inputValue as string)
    onChange(formattedDate as any)
    setIsEditing(false)
  }

  const startEditMode = () => {
    setIsEditing(true)
  }

  if (isEditing) {
    return (
      <>
        <DateInputContainer>
          <BirthdayTextInput
            type="datetime"
            placeholder={BIRTHDAY_FORMAT}
            placeholderTextColor={theme.common.inputPlaceholderColor}
            autoFocus={true}
            options={{
              format: 'MM/DD/YYYY',
            }}
            value={inputValue}
            onChangeText={handleTextChange}
            onEndEditing={handleEndEditing}
            keyboardType="numeric"
          />
        </DateInputContainer>
        <InfoText color={theme.common.inputTextError}>{error}</InfoText>
      </>
    )
  }

  const formatedBirthday = value ? formatBirthday(value as string) : ''

  return (
    <BirthdayButton onPress={startEditMode}>
      <ItemLabel>{value ? formatedBirthday : i18n.t('onb:birthday')}</ItemLabel>
    </BirthdayButton>
  )
}

export default BirthdayInput
