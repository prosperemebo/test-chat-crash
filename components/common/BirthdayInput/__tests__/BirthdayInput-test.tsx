import { render } from '@testing-library/react-native'
import { ThemeProvider } from 'styled-components/native'
import { describe, expect, test } from '@jest/globals'
import BirthdayInput from '../'
import { dark } from '@/theme'

describe('<BirthdayInput />', () => {

  test('BirthdayInput renders correctly', () => {
    const tree = render(
      <ThemeProvider theme={dark}>
        <BirthdayInput onChange={() => { }} value={new Date('2000-10-01')} />
        <BirthdayInput onChange={() => { }} value={new Date('2000-10-01')} error={'dsfs 34234'} />
      </ThemeProvider>,
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })
})