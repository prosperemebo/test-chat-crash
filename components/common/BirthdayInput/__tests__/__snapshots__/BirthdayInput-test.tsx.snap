// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BirthdayInput /> BirthdayInput renders correctly 1`] = `
[
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#15161A",
        "borderBottomLeftRadius": 8,
        "borderBottomRightRadius": 8,
        "borderColor": "#85899599",
        "borderTopLeftRadius": 8,
        "borderTopRightRadius": 8,
        "borderWidth": 0.5,
        "flexDirection": "row",
        "opacity": 1,
        "paddingBottom": 15,
        "paddingLeft": 15,
        "paddingRight": 15,
        "paddingTop": 15,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        {
          "color": "#FFFFFF",
          "flexBasis": 0,
          "flexGrow": 8,
          "flexShrink": 1,
          "fontFamily": "Poppins-Regular",
          "fontSize": 15,
          "marginLeft": 0,
          "paddingLeft": 0,
        }
      }
    >
      Oct 01, 2000
    </Text>
  </View>,
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#15161A",
        "borderBottomLeftRadius": 8,
        "borderBottomRightRadius": 8,
        "borderColor": "#85899599",
        "borderTopLeftRadius": 8,
        "borderTopRightRadius": 8,
        "borderWidth": 0.5,
        "flexDirection": "row",
        "opacity": 1,
        "paddingBottom": 15,
        "paddingLeft": 15,
        "paddingRight": 15,
        "paddingTop": 15,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        {
          "color": "#FFFFFF",
          "flexBasis": 0,
          "flexGrow": 8,
          "flexShrink": 1,
          "fontFamily": "Poppins-Regular",
          "fontSize": 15,
          "marginLeft": 0,
          "paddingLeft": 0,
        }
      }
    >
      Oct 01, 2000
    </Text>
  </View>,
]
`;
